src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -111,20 +112,21 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
     refreshPanelsVisibility()
 
     addWelcomeMessage()
-    refreshSubscriptionTab()
+    ApplicationManager.getApplication().executeOnPooledThread { refreshSubscriptionTab() }
     loadNewChatId()
   }
 
+  @RequiresBackgroundThread
   fun refreshSubscriptionTab() {
-    runInEdtAsync(Disposer.newCheckedDisposable()) {
-      tryRestartingAgentIfNotRunning(project)
-      getInitializedServer(project).thenAccept { server ->
-        if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
-          addNewSubscriptionTab(server)
-        } else {
+    tryRestartingAgentIfNotRunning(project)
+    getInitializedServer(project).thenAccept { server ->
+      if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
+        addNewSubscriptionTab(server)
```

Comment:
I looked into `addNewSubscriptionTab` and now it seems to be doing both long operations and UI operations.
Now long operations are on the background thread, but UI operations (like `tabbedPane.insertTab`) are not on the dispatch thread.